tests/components/homematicip_cloud/test_config_flow.py

```diff
@@ -1,34 +1,58 @@
 """Tests for HomematicIP Cloud config flow."""
-from unittest.mock import patch
+from asynctest import patch
 
-from homeassistant.components.homematicip_cloud import config_flow, const, hap as hmipc
+from homeassistant.components.homematicip_cloud import const
 
-from tests.common import MockConfigEntry, mock_coro
+from tests.common import MockConfigEntry
 
 
 async def test_flow_works(hass):
-    """Test config flow works."""
+    """Test config flow."""
     config = {
         const.HMIPC_HAPID: "ABC123",
         const.HMIPC_PIN: "123",
         const.HMIPC_NAME: "hmip",
     }
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
-
-    hap = hmipc.HomematicipAuth(hass, config)
-    with patch.object(hap, "get_auth", return_value=mock_coro()), patch.object(
-        hmipc.HomematicipAuth, "async_checkbutton", return_value=mock_coro(True)
-    ), patch.object(
-        hmipc.HomematicipAuth, "async_setup", return_value=mock_coro(True)
-    ), patch.object(
-        hmipc.HomematicipAuth, "async_register", return_value=mock_coro(True)
+
+    with patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_checkbutton",
+        return_value=False,
     ):
-        hap.authtoken = "ABC"
-        result = await flow.async_step_init(user_input=config)
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "user"}, data=config
+        )
 
-        assert hap.authtoken == "ABC"
-        assert result["type"] == "create_entry"
+    assert result["type"] == "form"
+    assert result["step_id"] == "link"
+    assert result["errors"] == {"base": "press_the_button"}
+
+    flow = next(
+        (
+            flow
+            for flow in hass.config_entries.flow.async_progress()
+            if flow["flow_id"] == result["flow_id"]
+        )
+    )
+    assert flow["context"]["unique_id"] == "ABC123"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)